src/components/UserUpdateGroups.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <v-card outlined class="subgroup" :disabled="!adminUser">
+    <h2>User Groups</h2>
+    <v-row no-gutters>
+      <v-col class="col-4">
+        <v-checkbox
+            hide-details="auto"
+            v-for="(group, index) in allGroups"
+            :id="'group-' + index"
+            v-model="selectedGroups"
+            :value="group"
+            :label="group.name"
+            :key="group.id"
+        ></v-checkbox>
+      </v-col>
+    </v-row>
+    <div class="my-6">
+      <v-btn id="save-user-groups" class="secondary"
+             medium v-on:click="updateUserGroups()"
+             v-if="adminUser">
+        Save User Groups
+      </v-btn>
+    </div>
+  </v-card>
+</template>
+
+<script>
+import UsersRepository from "@/api/UsersRepository";
+import GroupsRepository from "@/api/GroupsRepository";
+
+export default {
+  name: "UserUpdateGroups",
+  props: ['userId'],
+  data() {
+    return {
+      adminUser: false,
+      joinedGroups: [],
+      allGroups: [],
+      selectedGroups: []
+    };
+  },
+  async created() {
+    this.checkIfAdmin();
+    await this.getUserGroupsData();
+  },
+  methods: {
+    checkIfAdmin: function() {
+      if (this.$keycloak.tokenParsed.resource_access['user-management'] &&
+          this.$keycloak.tokenParsed.resource_access['user-management'].roles.includes('user-management-admin')) {
+        this.adminUser = true;
+      }
+    },
+    getUserGroupsData: async function() {
+      this.joinedGroups = [];
+      this.allGroups = [];
+      this.selectedGroups = [];
+
+      let userGroupResponses = await Promise.all([UsersRepository.getUserGroups(this.userId), GroupsRepository.get()])
+          .catch(error => {
+            this.$store.commit("alert/setAlert", {
+              message: "Error updating loading user groups: " + error,
+              type: "error"
+            });
+          });
+
+      this.joinedGroups.push(...userGroupResponses[0].data);
+      this.selectedGroups.push(...userGroupResponses[0].data);
+      this.allGroups.push(...userGroupResponses[1].data);
+      // Keycloak "Groups" API returns a different object structure than users/groups
+      // We need them to match for the checkboxes to map
+      this.allGroups = this.allGroups.map(({id,name,path})=>({id,name,path}));
```

Comment:
`map(({id,name,path})->({id,name,path})) looks like it does nothing? Or is it discarding some attributes in the object?

Reply:
It's discarding a "subgroups" attribute in the object.

Reply:
Does that matter?

Reply:
yes the vuetify check boxes expects the object structure to match otherwise it won't set any of the check boxes to "checked"